### PR TITLE
Avoid duplicate "warning:" or "error:" output

### PR DESCRIPTION
**Pull request by tautschnig**
The gcc message handler already took care of this; make the console message
handler do the same.

### CODE REVIEW

**Comment by kroening** [comment]
Uhoh, lots of regressions check for "CONVERSION ERROR". Not clear whether to change the tests or do a special case for that.

**Comment by tautschnig** [comment]
Assigning to myself for further work at a later time. No urgency/dependency here for anyone.

**Comment by tautschnig** [comment]
> Uhoh, lots of regressions check for "CONVERSION ERROR". Not clear whether to change the tests or do a special case for that.

I have revamped this PR, and none of the "CONVERSION ERROR" messages should be affected. It's really just to avoid "warning: warning: " being printed to users.

**Review by martin-cs — APPROVED** [review]
Much respect for resolving a 2017 issue!

**Review by peterschrammel — APPROVED** [review]

**Comment by NlightNFotis** [comment]
Hi @tautschnig, would it be possible to rebase this on top of current `develop` so that the CI checks get updated?

**Comment by tautschnig** [comment]
> Hi @tautschnig, would it be possible to rebase this on top of current `develop` so that the CI checks get updated?

Done!

**Comment by NlightNFotis** [comment]
Hi @tautschnig, could you push this one again please?

It seems one of the CI checks has failed to initialise, blocking the whole PR from getting merged despite approvals.

**Comment by TGWDB** [comment]
@tautschnig Looks like this would probably pass if it was pushed with a tiny change to rerun CI jobs?

**Review by TGWDB — APPROVED** [review]

**Comment by TGWDB** [comment]
Closing due to age (no further comment on PR content), please reopen with rebase on develop if you intent to continue this work.

**Comment by tautschnig** [comment]
Re-opening: this is just waiting for a review.

**Comment by TGWDB** [comment]
@remi-delmas-3000 @kroening Woulda review be possible?

File: regression/ansi-c/linking_conflicts1/test.desc [hunk 1]
```diff
@@ -3,7 +3,7 @@ main.c
 other.c
 ^EXIT=(64|1)$
 ^SIGNAL=0$
-error: conflicting function declarations 'bar'
-error: conflicting function declarations 'bar2'
+conflicting function declarations 'bar'
+conflicting function declarations 'bar2'
 --
```
Comment (kroening) [inline regression/ansi-c/linking_conflicts1/test.desc:8]: Make that $error: conflicting...., so it actually tests that the error isn't there twice.

File: regression/cbmc/incomplete-structs/test.desc [hunk 2]
```diff
@@ -1,8 +1,8 @@
 CORE
 typesmain.c
 types1.c types2.c types3.c
-warning: pointer parameter types differ between declaration and definition "bar"
-warning: pointer parameter types differ between declaration and definition "foo"
+pointer parameter types differ between declaration and definition 'bar'
+pointer parameter types differ between declaration and definition 'foo'
```
Comment (kroening) [inline regression/cbmc/incomplete-structs/test.desc]: same here

File: src/goto-cc/cl_message_handler.h [hunk 3]
```diff
@@ -29,6 +29,16 @@ class cl_message_handlert : public console_message_handlert
     const source_locationt &location) override;
 
   using console_message_handlert::print;
+
+  /// With \p yes set to \c true, prefix warnings with an error message.
+  /// \param yes: Whether or not to prefix warnings.
+  void print_warnings_as_errors(bool yes)
```
Comment (kroening) [inline src/goto-cc/cl_message_handler.h:35]: Do we really need that?

Comment (tautschnig) [reply src/goto-cc/cl_message_handler.h:35]: I'd say it depends on how closely we'd like to mimic the behaviour of Visual Studio and GCC. Prior to this change, we just printed "warning" but nonetheless failed with a non-zero exit code when `Werror` was used. With this change, we would actually print "error" for such now-fatal warnings, as those compilers do.

File: regression/ansi-c/linking_conflicts1/test.desc (hunk 1 above)
Comment (tautschnig) [reply regression/ansi-c/linking_conflicts1/test.desc:8]: Amended.

File: regression/cbmc/incomplete-structs/test.desc (hunk 2 above)
Comment (tautschnig) [reply regression/cbmc/incomplete-structs/test.desc]: Amended.

File: src/goto-cc/cl_message_handler.h (hunk 3 above)
Comment (kroening) [reply src/goto-cc/cl_message_handler.h:35]: Feels a bit cosmetic, but why not.

**Review by kroening — APPROVED** [review]

**Pull request merged**